langchain/src/vectorstores/vectara.ts

```diff
@@ -0,0 +1,250 @@
+          body: JSON.stringify(data),
+        });
+        const result = await response.json();
+        if (
+          result.status?.code !== "OK" &&
+          result.status?.code !== "ALREADY_EXISTS"
+        ) {
+          return {
+            code: 500,
+            detail: `Vectara API returned status code ${result.code}: ${result.message}`,
+          };
+        } else {
+          countAdded += 1;
+        }
+      } catch (e) {
+        return {
+          code: 500,
+          detail: `Error ${e} while adding document ${document}`,
+        };
+      }
+    }
+    return {
+      code: 200,
+      detail: `Added ${countAdded} documents to Vectara`,
+    };
+  }
+
+  async similaritySearchWithScore(
+    query: string,
+    k = 10,
+    filter: VectaraFilter | undefined = undefined
+  ): Promise<[Document, number][]> {
+    const headers = await this.getJsonHeader();
+    const data = {
+      query: [
+        {
+          query: query,
+          numResults: k,
+          corpusKey: [
+            {
+              customerId: this.customer_id,
+              corpusId: this.corpus_id,
+              metadataFilter: filter?.filter ?? "",
+              lexicalInterpolationConfig: { lambda: filter?.lambda ?? 0.025 },
+            },
+          ],
+        },
+      ],
+    };
+
+    try {
+      const response = await fetch(`https://${this.api_endpoint}/v1/query`, {
+        method: "POST",
+        headers: headers?.headers,
+        body: JSON.stringify(data),
+      });
+      if (response.status !== 200) {
+        throw new Error(`Vectara API returned status code ${response.status}`);
+      }
+      const result = await response.json();
+      const responses = result.responseSet[0].response;
+      const documentsAndScores = responses.map((response: any) => {
+        return [
+          new Document({
+            pageContent: response.text,
+            metadata: response.metadata,
+          }),
+          response.score,
+        ];
+      });
+      return documentsAndScores;
+    } catch (e) {
+      throw e;
+    }
+  }
+
+  async similaritySearch(
+    query: string,
+    k = 10,
+    filter: VectaraFilter | undefined = undefined
+  ): Promise<Document[]> {
+    const resultWithScore = await this.similaritySearchWithScore(
+      query,
+      k,
+      filter
+    );
+    return resultWithScore.map((result) => result[0]);
+  }
+
+  async similaritySearchVectorWithScore(
+    _query: number[],
+    _k: number,
+    _filter?: VectaraFilter | undefined
+  ): Promise<[Document<Record<string, any>>, number][]> {
+    throw new Error(
+      "Method not implemented. Please call similaritySearch or similaritySearchWithScore instead."
+    );
+  }
+
+  static fromTexts(
+    texts: string[],
+    metadatas: object | object[],
+    embeddings: Embeddings,
+    args: VectaraLibArgs
+  ): Promise<VectaraStore> {
+    const embeddingsName = embeddings.constructor.name;
+    if (embeddingsName !== "FakeEmbeddings") {
```

Comment:
This will unfortunately fail mysteriously in minified environments, so best not to include. Docs should be enough for this.